system-parachains/coretime/coretime-kusama/src/coretime.rs

```diff
@@ -64,13 +69,52 @@ parameter_types! {
 
 /// Burn revenue from coretime sales. See
 /// [RFC-010](https://polkadot-fellows.github.io/RFCs/approved/0010-burn-coretime-revenue.html).
-pub struct BurnRevenue;
-impl OnUnbalanced<Credit<AccountId, Balances>> for BurnRevenue {
-	fn on_nonzero_unbalanced(credit: Credit<AccountId, Balances>) {
-		let _ = <Balances as Balanced<_>>::resolve(&CoretimeBurnAccount::get(), credit);
+pub struct BurnCoretimeRevenue;
+impl OnUnbalanced<Credit<AccountId, Balances>> for BurnCoretimeRevenue {
+	fn on_nonzero_unbalanced(amount: Credit<AccountId, Balances>) {
+		let acc = CoretimeBurnAccount::get();
+		if !System::<Runtime>::account_exists(&acc) {
+			System::<Runtime>::inc_providers(&acc);
+		}
+		Balances::resolve(&acc, amount).defensive_ok();
 	}
 }
 
+type AssetTransactor = <xcm_config::XcmConfig as xcm_executor::Config>::AssetTransactor;
+
+fn burn_at_relay(stash: &AccountId, value: Balance) -> Result<(), XcmError> {
+	let dest = Location::parent();
+	let stash_location =
+		Junction::AccountId32 { network: None, id: stash.clone().into() }.into_location();
+	let asset = Asset { id: AssetId(Location::parent()), fun: Fungible(value) };
+	let dummy_xcm_context = XcmContext { origin: None, message_id: [0; 32], topic: None };
+
+	let withdrawn = AssetTransactor::withdraw_asset(&asset, &stash_location, None)?;
+
+	AssetTransactor::can_check_out(&dest, &asset, &dummy_xcm_context)?;
```

Comment:
we need some common utility to perform such operations, should we leave todo here

Reply:
Sounds like you want to have an issue and not a TODO?

Reply:
As an author I would take a note first, and try to solve this with `pallet_xcm::transfer_assets_using_type_and_then` or `InitiateTeleport`. If not applicable, I would check with xcm team, I think we have something planned for such cases. If not, I would create an issue.

Reply:
https://github.com/polkadot-fellows/runtimes/issues/404